### PR TITLE
feat: add sourcemap for postcss transform

### PR DESCRIPTION
**Pull request by ahabhgk**

### CODE REVIEW

File: crates/turbopack-core/src/source_transform.rs [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
 pub trait SourceTransform {
-    fn transform(&self, source: AssetVc) -> AssetVc;
+    fn transform(&self, source: SourceTransformedVc) -> SourceTransformedVc;
+}
+
+#[turbo_tasks::value]
+pub struct SourceTransformed {
+    pub source: AssetVc,
+    pub source_map: Option<SourceMapVc>,
+}
```
Comment (sokra) [inline crates/turbopack-core/src/source_transform.rs:14]: I wonder if our `AssetVc` should have a `source_map(): OptionSourceMapVc` instead?

Comment (sokra) [reply crates/turbopack-core/src/source_transform.rs:14]: Or our `AssetContent` should have a `source_map: Option<SourceMapVc>` field?

Comment (ahabhgk) [reply crates/turbopack-core/src/source_transform.rs:14]: Both way seems hold a source map on `AssetVc`, if so should `AssetVc::source_map` or `AssetContent.source_map` needs also generate a "output source map" (`CssChunkSourceMapAsset`), currently the output source map needs implement `Asset` (`impl Asset for CssChunkSourceMapAsset`), or is it only suitable for "input source map" (source map generated by loader, source map in node_modules)?

For this two option I would choose `AssetVc::source_map(): OptionSourceMapVc`, source map can remain lazily generate, not sure `AssetContent.source_map` could do that.

Comment (sokra) [reply crates/turbopack-core/src/source_transform.rs:14]: Just remembered... We already have a `GenerateSourceMap` trait that one can optionally implement to allow to get a source map for an Asset. So I just you could implement that trait on the `PostCssTransformedAsset` to allow to get a source map.

Comment (ahabhgk) [reply crates/turbopack-core/src/source_transform.rs:14]: Tried to implement `GenerateSourceMap` for `PostCssTransformedAsset` before, but the problem is source map could be `None` when error happened in postcss transforming, and not all loaders must return a source map (off-topic from this PR), but `GenerateSourceMap` can't return a `None` for source map.

Comment (ahabhgk) [reply crates/turbopack-core/src/source_transform.rs:14]: For "optionally implement" do you mean optionally implement `GenerateSourceMap` for different Asset based on the transforming result, for example

```rust
enum ProcessPostCssResult {
  NotFound(PostcssNotFoundAssetVc),
  Transformed(PostCssTransformedAssetVc),
}
impl GenerateSourceMap for PostCssTransformedAssetVc {/* ... */}
impl Asset for PostCssTransformedAssetVc {/* ... */}
impl Asset for PostcssNotFoundAssetVc {/* ... */}
```

**Comment by anthonyshew** [comment]
Turbopack has moved! https://vercel.com/blog/turbopack-moving-homes-6p1uPy74FtuOo2Ca8VPqOW

**Pull request closed without merging**